### DISCUSSION TITLE
Block Plugin in OnlyOffice

### DISCUSSION THREAD

**Discussion by IwisGiwis** [SafeExamBrowser/seb-win-refactoring#394] [Q&A] [answered]
Dear all,
at my university we are trying to implement the use of SEB for taking on-campus digital exams. Some lecturers want the students to use Only Office for exams in OLAT. But Only Office has plugins like Youtube, which can be used for cheating. Is it possible to tweak some settings in the configuration file to block the URL of youtube in OnlyOffice? 

Best regards.
Stefan

**Comment by dbuechel** [comment, marked as answer]
Yes, you can block URLs by using the URL filter functionality of SEB: https://safeexambrowser.org/windows/win_usermanual_en.html#NetworkPane

**Reply by IwisGiwis** [reply]
I know that you can block URLs but it seems that it does not work with plugins in OnlyOffice.  
Here are my filters where i block the URL https://www.youtube.com/watch?v=pQrsGacFwbk  
![confi_file](https://user-images.githubusercontent.com/74247647/162901604-022c76e4-41ee-499f-aab3-dc32ab31b801.png)  
But i can still open it in OnlyOffice
![only_office](https://user-images.githubusercontent.com/74247647/162903273-9d69a922-a4b7-4727-9463-65c3ec147953.png)

**Reply by dbuechel** [reply]
That is because embedded YouTube videos have a different URL, in that specific case https://www.youtube.com/embed/pQrsGacFwbk.

**Reply by IwisGiwis** [reply]
Thanks for the quick reply!  
I know that it sounds weird, but it worked once and when i tried to block every video with https://www.youtube.com/embed/* it did not work. When i changed it back to https://www.youtube.com/embed/pQrsGacFwbk it no longer works. I tried it with several other rules but that does not help either.  
![filter](https://user-images.githubusercontent.com/74247647/162920714-16600076-a50b-4101-8bfe-b8b3bc859b13.png)  
Besides shouldn't the regular expression 'youtube' be enough to filter all URLs with youtube as a substring? We are using version 3.4.0 beta. Maybe it's a bug?

**Reply by dbuechel** [reply]
URL filter rules (and especially regex) are quite complicated indeed, but they should work as specified (all unit tests are at least completing correctly). I'd activate the live application log (see [documentation](https://safeexambrowser.org/windows/win_usermanual_en.html#SecurityPane) -> "Allow access to application log"), this way you can test and verify your settings in real time while you're accessing various sites in SEB as any blocked URLs are reported in the application log.

**Reply by IwisGiwis** [reply]
Sorry for the wall of text.
I activated the live application log and when i open a link, the application log shows "Initialized request filter with 5 rule(s)". But not when i enter a URL in the youtube plugin. I found a youtube video that always gets blocked. Maybe because of the format. Not sure to be honest. 
The manual also says: If no allow rule matched, then the request is discarded.  
I have no allow rule for youtube. So everything should be blocked.
In OnlyOffice you can also import entire websites via Compare in Collaboration. The application log doesn't show "Initialized request filter with 5 rule(s)". So i assume filtering of URLs in OnlyOffice doesn't work as intendend at least on our site.
![compare](https://user-images.githubusercontent.com/74247647/163144983-75caf374-b633-46af-b4a4-54fdfc6b47fa.png)

**Reply by dbuechel** [reply]
Okay, my bad: I just realized that you're using a standalone third-party application. The URL filter only applies for the integrated browser of SEB itself, not for third-party applications or other (system) components.

**Reply by IwisGiwis** [reply]
No problem and thanks for the reply. At least i know now that it's not possible. 
Thank you for the support! :)
